include/irr/asset/filters/CMatchedSizeInOutImageFilterCommon.h

```diff
@@ -28,6 +46,19 @@ class CMatchedSizeInOutImageFilterCommon : public CBasicImageFilterCommon
 				}
 				virtual ~CState() {}
 
+				/*
+					Wrapped extent and layer count to an union.
+					You can fill image extent and layer count
+					separately or fill it at once by \bextentLayerCount\b,
+					where in it \bx, y, z\b is treated as extent, and a \bw\b 
+					is treated as layerCount. You can use it interchangeably.
+
+					Pay attention output image must be prepared to conversion
+					process. It means it's a user resposibility to take care
+					of attached regions and texel buffer with new adjusted size
+					for executing the process for output image.
```

Comment:
actually in most filters there is a `default_clip_functor_t` which takes care of this (shrinks the extents and offsets to fit individual regions, regions are processed from first to last independently)

Reply:
only restriction is that the offsets + extents cannot specify an area outside of the whole image itself